### PR TITLE
Tile layer support

### PR DESCRIPTION
**Pull request by taeymma**
Added support for DeckGL Tile Layer.

### CODE REVIEW

File: src/layers/base-layer.ts [hunk 1]
```diff
@@ -852,6 +852,9 @@ class Layer {
    * @returns {boolean} yes or no
    */
   hasLayerData(layerData) {
+    if (this.type == 'tile' && Boolean(layerData.url && layerData.url.length)) {
```
Comment (Polygens) [inline src/layers/base-layer.ts]:
```suggestion
    if (this.type === 'tile' && Boolean(layerData.url && layerData.url.length)) {
```

File: src/layers/tile-layer/tile-layer.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        maxZoom: 19,
+        tileSize: 256,
+
+        renderSubLayers: props => {
+          const {
+            bbox: {west, south, east, north}
+          } = props.tile;
+
+          return new BitmapLayer(props, {
+            data: null,
```
Comment (Polygens) [inline src/layers/tile-layer/tile-layer.ts:113]: Can't we leave this field out?

File: src/layers/tile-layer/tile-layer.ts [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    const {data, gpuFilter, mapState, animationConfig} = opts;
+    const {url} = data;
+
+    // Create new deck.gl Tile Layer with Bitmap sublayers
+    return [
+      new DeckGLTileLayer({
+        data: url,
+        minZoom: 0,
+        maxZoom: 19,
+        tileSize: 256,
```
Comment (Polygens) [inline src/layers/tile-layer/tile-layer.ts]: Would use the deckGL default of 512 to reduce http request latency

File: src/layers/tile-layer/tile-layer.ts [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+  formatLayerData(datasets, oldLayerData) {
+    const {gpuFilter, dataContainer} = datasets[this.config.dataId];
+    // const {data} = this.updateData(datasets, oldLayerData);
+    const accessors = this.getAttributeAccessors({dataContainer});
+    // extract URL from row 0 column 0
+    const url = dataContainer.valueAt(0, 0);
+    return {
+      url,
+      getFilterValue: gpuFilter.filterValueAccessor(dataContainer)(),
```
Comment (Polygens) [inline src/layers/tile-layer/tile-layer.ts]: We don't use gpu filters here I suppose? Probably not even possible to combine with our bitmaplayer approach. You can leave it in for the PoC, just wondering

**Review by Polygens — APPROVED** [review]
Added some remarks, nothing major so approved

File: src/layers/tile-layer/tile-layer.ts (hunk 4 above)
Comment (taeymma) [reply src/layers/tile-layer/tile-layer.ts]: This was based on one of the other layers, so might be just copied ;)

File: src/layers/tile-layer/tile-layer.ts (hunk 2 above)
Comment (taeymma) [reply src/layers/tile-layer/tile-layer.ts:113]: DeckGL seems to crash when I leave this out. It probably expects `null` and not `undefined`

**Pull request merged**